src/idlcxx/tests/generator/re_alignment_3_impl.cpp.in

```diff
@@ -0,0 +1,146 @@
+#include "org/eclipse/cyclonedds/topic/hash.hpp"
+
+size_t s::write_struct(void *data, size_t position) const
+{
+  for (size_t _i1 = 0; _i1 < 3; _i1++)  {
+    size_t _al1 = (4 - (position&0x3))&0x3;  //alignment
+    memset(static_cast<char*>(data)+position,0x0,_al1);  //setting alignment bytes to 0x0
+    position += _al1;  //moving position indicator
+    uint32_t _se1 = static_cast<uint32_t>(string_arr()[_i1].size()+1);  //number of entries in the sequence
+    *reinterpret_cast<uint32_t*>(static_cast<char*>(data) + position) = _se1;  //writing entries for member: string_arr()[_i1]
+    position += 4;  //moving position indicator
+    memcpy(static_cast<char*>(data)+position,string_arr()[_i1].data(),_se1*1); //writing bytes for member: string_arr()[_i1]
+    position += _se1;  //entries of sequence
+    size_t _al1 = (4 - (position&0x3))&0x3;  //alignment
```

Comment:
@reicheratwork With the current changes, I get a re-declaration error of the alignment variable. Same as here, previously declared in line 6

Reply:
@sumanth-nirmal you are correct, something is going wrong in the alignment algorithm.

I have created and pushed a fix.